### ISSUE TITLE
Can you update Autobalance for the latest TC?

### ISSUE THREAD

**Issue by jasanhell1** [trickerer/Autobalance_for_NPCBots#6]
I understand many people are using AC now, but can you please update the autobalance for the latest TC?

**Comment by trickerer** [comment]
Oh-h... 16 months since last update, huh. Forgot it even existed.
Well I can, once I'm done with TC bots update.
Probably can't be a quick fix. Does it give many compilations errors?

**Comment by jasanhell1** [comment]
I did not use the latest TC. I used your repo TrinityCore-3.3.5-with-NPCBots.
No too many compliations errors. 

**1) I manually add the code to ScriptMgr.h.**

patching file src/server/game/Scripting/ScriptMgr.h
Hunk # 1 FAILED at 309.
Hunk # 2 succeeded at 393 with fuzz 2 (offset -6 lines).
Hunk # 3 succeeded at 423 with fuzz 1 (offset -6 lines).
Hunk # 4 succeeded at 962 with fuzz 2 (offset 11 lines).
Hunk # 5 succeeded at 1107 (offset 13 lines).
1 out of 5 hunks FAILED -- saving rejects to file src/server/game/Scripting/ScriptMgr.h.rej

class AllMapScript : public ScriptObject
{
protected:
    AllMapScript(const char* name);

public:
    // Called when player enters any map
    virtual void OnPlayerEnterAll(Map* /*map*/, Player* /*player*/) { }

    // Called when player leaves any map
    virtual void OnPlayerLeaveAll(Map* /*map*/, Player* /*player*/) { }

    virtual void OnAllUpdate(Map* /*map*/, uint32 /*diff*/) { }
};

**2) change AutoBalance.cpp, MAXGROUPSIZE to MAX_GROUP_SIZE.**

uint32 maxNumberOfPlayers = instanceMap ? instanceMap->GetMaxPlayers() : MAX_GROUP_SIZE;

Then, the compliation can be sucessed. 
I did not test, but I think it may need to update for the latest TC. 
TC team has changed Scripting format.
